logger.js

```diff
@@ -59,12 +59,12 @@ module.exports = function(config) {
 	};
 
 	function snipFragileData(data) {
-		for (const key in data) {
+		return Object.keys(data).map(key => {
 			if (key.search(/passphrase|password/i) > -1) {
 				data[key] = 'XXXXXXXXXX';
 			}
-		}
-		return data;
+			return data[key];
+		});
 	}
```

Comment:
This new function is not equivalent to the old one. It returns an array rather than an object.
In order to make it equivalent you have to either return the mutated object or reconstruct it using `reduce`.
As the motivation to not use iterators is to have pure functions, I would suggest using `reduce`

```
function snipFragileData(data) {
	return Object.keys(data).reduce((acc, key) => {
		acc[key] = (key.search(/passphrase|password/i) > -1) ? 'XXXXXXXXXX' : data[key];
		return acc;
	}, {});
}
```